### PR TITLE
FileWriter: support for setting file modification time via mtime option

### LINKED ISSUE #169: CAF::FileWriter support mtime

### PR DESCRIPTION
**Pull request by stdweird**
Fixes #169

### CODE REVIEW

File: src/test/perl/test-caffilewriter.t [hunk 1]
```diff
@@ -124,13 +124,18 @@ $fh = CAF::FileWriter->open (FILENAME, log => $this_app,
 			     backup => "foo",
 			     mode => 0400,
 			     owner => 100,
-			     group => 200);
+			     group => 200,
+                 mtime => 1234567);
```
Comment (ned21) [inline src/test/perl/test-caffilewriter.t]: spacing issue?

Comment (stdweird) [inline src/test/perl/test-caffilewriter.t]: since https://github.com/quattor/CAF/pull/181 also modifies the test files, i made a bunch of whitespace cleanup in that PR
in this case https://github.com/quattor/CAF/pull/181/files#diff-0c4858928999b3cebc2f0cd9a75201a1R143
